metric/common.go

```diff
@@ -54,9 +54,26 @@ type baseEntry interface {
 	read(t time.Time) metricdata.Point
 }
 
+func (bm *baseMetric) startTime() time.Time {
+	switch bm.bmType {
+	case cumulativeInt64:
+		return bm.start
+	case cumulativeFloat64:
+		return bm.start
+	case derivedCumulativeInt64:
+		return bm.start
+	case derivedCumulativeFloat64:
+		return bm.start
+	default:
+		// gauges don't have start time.
+		return time.Now()
+	}
+}
+
 // Read returns the current values of the baseMetric as a metric for export.
 func (bm *baseMetric) read() *metricdata.Metric {
 	now := time.Now()
+	startTime := bm.startTime()
```

Comment:
`if startTime == nil { startTime = now }` so that we can guarantee startTime and now are the same.

Reply:
done.